packages/dds/tree/docs/undo/README.md

```diff
@@ -169,6 +170,139 @@ Partial undo would allow such a user to only undo the changes within the region
 This could be achieved by including in the undo message a characterization of the region of the document to be affected by the undo.
 This may need to be characterized as a combination of input context regions and output context regions.
 
+## V1 Undo
+
+Here we detail the vision for the first implementation of undo/redo.
+This implementation is meant to satisfy our needs for parity with experimental (AKA legacy) SharedTree.
+
+This first version aims to achieve some basic undo functionality with a minimum amount of code changes and complexity.
+To that end, we mostly reuse the existing code paths for changesets by always sending
+[concrete undos](#abstract-vs-concrete-undo-messages) over the wire.
+The undo edit is created by inverting the edit that needs to be undone,
+and rebasing that inverse over all the changes that have been applied since.
+
+Sending concrete undo edits alleviates the need establish and maintain distributed consensus on an undo window.
+It also means summaries do not need to include repair data.
+It does however require sending rebased changes over the wire.
+
+### Concrete Undo of Unsequenced Changes
+
+Using concrete undos even when the change to be undone has not been sequence is somewhat problematic
+because we cannot know in advance the exact impact of the change to undo.
+
+For this V1, we simply produce a "best attempt" undo based on the most up to date version of the change to be undone.
+This could lead to some data loss in scenarios where the change to be undone deletes a subtree under which content is concurrently inserted.
+
+#### Creating Concrete Redo Edits
+
+Redo changesets should be created by inverting the corresponding undo changeset and rebasing that inverse all the edits that were applied since the undo.
```

Comment:
```suggestion
Redo changesets should be created by inverting the corresponding undo changeset and rebasing that inverse over all the edits that were applied since the undo.
```